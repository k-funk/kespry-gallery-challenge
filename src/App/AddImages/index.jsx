import React, { useState } from 'react';
import { PropTypes as T } from 'prop-types';


const AddImages = ({ addImage }) => {
  const [imgUrl, onChange] = useState('');

  const onSubmit = e => {
    e.preventDefault();
    addImage(imgUrl);
    onChange('');
  };

  return (
    <div className="card">
      <h4>Add Image URLs</h4>
      <form onSubmit={onSubmit}>
        <input
          type="url"
          placeholder="https://picsum.photos/seed/picsum/450/300"
          onChange={e => onChange(e.target.value)}
          value={imgUrl}
        />
        <button className="btn" type="submit">
          Add Image
        </button>
      </form>
    </div>
  );
};

AddImages.propTypes = {
  addImage: T.func.isRequired,
};

export default AddImages;
