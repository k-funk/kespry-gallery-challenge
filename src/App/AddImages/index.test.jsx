import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import AddImages from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let addImageSpy;

  beforeEach(() => {
    addImageSpy = jest.fn();
  });

  test('(default)', () => {
    wrapper = shallow((
      <AddImages addImage={addImageSpy} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
