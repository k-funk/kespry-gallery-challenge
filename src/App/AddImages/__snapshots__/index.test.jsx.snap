// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `
<div
  className="card"
>
  <h4>
    Add Image URLs
  </h4>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="https://picsum.photos/seed/picsum/450/300"
      type="url"
      value=""
    />
    <button
      className="btn"
      type="submit"
    >
      Add Image
    </button>
  </form>
</div>
`;
